src/app/clusters/ethernet_network_diagnostics_server/ethernet_network_diagnostics_server.cpp

```diff
@@ -16,11 +16,106 @@
  */
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterceptor.h>
 #include <app/CommandHandler.h>
+#include <app/MessageDef/AttributeDataElement.h>
 #include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/Optional.h>
+#include <platform/ConnectivityManager.h>
 
 using namespace chip;
+using namespace chip::app;
 using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EthernetNetworkDiagnostics::Attributes;
+using chip::DeviceLayer::ConnectivityManager;
+
+namespace {
+
+class EthernetDiagosticsAttrInterceptor : public AttributeAccessInterceptor
+{
+public:
+    // Register for the EthernetNetworkDiagnostics cluster on all endpoints.
+    EthernetDiagosticsAttrInterceptor() :
+        AttributeAccessInterceptor(Optional<EndpointId>::Missing(), EthernetNetworkDiagnostics::Id)
+    {}
+
+    CHIP_ERROR ReadAttributeData(ClusterInfo & aClusterInfo, TLV::TLVWriter * aWriter, bool * aDataRead) override;
+
+    bool IsRegistered() const { return sIsRegistered; }
+
+    void RegisterIfNeeded()
+    {
+        if (!sIsRegistered)
+        {
+            registerAttributeAccessInterceptor(this);
+        }
+        sIsRegistered = true;
+    }
+
+private:
+    static bool sIsRegistered;
+
+    CHIP_ERROR ReadDataIfSupported(CHIP_ERROR (ConnectivityManager::*getter)(uint64_t &), TLV::TLVWriter * aWriter);
+};
+
+EthernetDiagosticsAttrInterceptor gAttrInterceptor;
+bool EthernetDiagosticsAttrInterceptor::sIsRegistered = false;
+
+CHIP_ERROR EthernetDiagosticsAttrInterceptor::ReadAttributeData(ClusterInfo & aClusterInfo, TLV::TLVWriter * aWriter,
```

Comment:
As designed so far, it looks like these read accessor methods are only called when handling an IM Read, or the initial set of reports in an IM Subscribe.

I don't think this PR addresses the use-case of reporting post subscription establishment, where there needs to be some mechanism in this cluster impl to signal change in attribute values to the IM Engine.

Follow-up PR is fine, but that does change the implementation here to not just be a pull, but a push-based model as well.

Reply:
That's right, just like for external attributes we need some way to notify the stack when values (might) have changed.